Sources/Styleguide/DesignSystems/TypeScale.swift

```diff
@@ -1,13 +1,61 @@
 import Css
 import Prelude
 
+public enum TypeScale: String {
+  case r4
+  case r3
+  case r2
+  case r1_5
+  case r1_25
+  case r1
+  case r0_875
+  case r0_75
+
+  fileprivate var size: Size {
+    switch self {
+    case .r4:
+      return .rem(4)
+    case .r3:
+      return .rem(3)
+    case .r2:
+      return .rem(2)
+    case .r1_5:
+      return .rem(1.5)
+    case .r1_25:
+      return .rem(1.25)
+    case .r1:
+      return .rem(1)
+    case .r0_875:
+      return .rem(0.875)
+    case .r0_75:
+      return .rem(0.75)
+    }
+  }
+
+  fileprivate static let allTypeScales: [TypeScale] = [.r4, .r3, .r2, .r1_5, .r1_25, .r1, .r0_875, .r0_75]
+}
+
 extension Class {
   public static let h1 = CssSelector.class("h1")
   public static let h2 = CssSelector.class("h2")
   public static let h3 = CssSelector.class("h3")
   public static let h4 = CssSelector.class("h4")
   public static let h5 = CssSelector.class("h5")
   public static let h6 = CssSelector.class("h6")
+
+  public static func typeScale(_ data: [Breakpoint: TypeScale]) -> CssSelector {
```

Comment:
this is the new type scale system. you can use it like:

```swift
h1(
  [`class`([Class.typeScale([.mobile: .r1, .desktop: r1_5])],
  ["Point-Free"]
)
```

That will have `1rem` on mobile and `1.5rem` on desktop. I was really annoyed at how the other type scale had numbers that didnt correlate to how many `rem`s there were, i.e. `h1` was `4rem`, `h2` was `3rem`, etc...